client/src/controllers/AutoSizeController.test.js

```diff
@@ -0,0 +1,77 @@
+import { Application } from '@hotwired/stimulus';
+import autosize from 'autosize';
+import { AutoSizeController } from './AutoSizeController';
+
+jest.mock('autosize');
+
+describe('AutoSizeController', () => {
+  let application;
+
+  beforeEach(() => {
+    application = Application.start();
+    application.register('w-auto-size', AutoSizeController);
+    document.body.innerHTML = `
+    <textarea 
+    class="w-field__autosize" 
+    data-controller="w-auto-size"
+    >
+    </textarea>
+    `;
+  });
+
+  afterEach(() => {
+    application.stop();
+  });
+
+  it('calls autosize on connect', () => {
+    jest.clearAllMocks();
+    const textarea = document.querySelector('[data-controller="w-auto-size"]');
+    expect(autosize).not.toHaveBeenCalled();
+    const controller = application.getControllerForElementAndIdentifier(
+      textarea,
+      'w-auto-size',
+    );
+    controller.connect();
+    expect(autosize).toHaveBeenCalledWith(textarea);
+  });
+
+  it('calls autosize.destroy on disconnect', () => {
+    const textarea = document.querySelector('[data-controller="w-auto-size"]');
+    const controller = application.getControllerForElementAndIdentifier(
+      textarea,
+      'w-auto-size',
+    );
+    controller.connect();
+    expect(autosize.destroy).not.toHaveBeenCalled();
+    controller.disconnect();
```

Comment:
Hey,

Sorry if it was not clear but you should never have to manually call `disconnect`, this is a callback method that runs before the controlled element is removed from the DOM.

Instead, run your test by removing the element from the DOM, then check that autosize.destroy was called.

```
const textarea = document.querySelector('[data-controller="w-auto-size"]');
expect(autosize.destroy).not.toHaveBeenCalled();
textarea.remove();
expect(autosize.destroy).toHaveBeenCalledWith(textarea);
```

You should also never manually call `connect`, this happens by default when Stimulus connects itself to DOM element.